verilog/analysis/checkers/dff_name_style_rule.cc

```diff
@@ -0,0 +1,402 @@
+    //  data_q2 <= data_ff -> WRONG
+    if (rhs_str != expected_rhs) {
+      violations_.insert(LintViolation(
+          rhs_expr, absl::StrCat(rhs_str, " Should be ", expected_rhs),
+          context));
+    }
+    return;
+  }
+
+  absl::string_view rhs_base =
+      CheckSuffix(context, rhs_expr, rhs_str, valid_input_suffixes);
+
+  // If the rhs is wrongly formatted, there is no need to check that the
+  // bases match
+  if (rhs_base.empty()) return;
+
+  if (lhs_base != rhs_base) {
+    // Bases should be equal
+    //  "a_q <= a_n" -> OK
+    //  "a_q <= b_n" -> WRONG
+    violations_.insert(LintViolation(
+        non_blocking_assignment,
+        absl::StrCat("Both parts before the suffix should be equal, but \"",
+                     lhs_base, "\" != \"", rhs_base, "\""),
+        context));
+  }
+}
+
+void DffNameStyleRule::HandleSymbol(const verible::Symbol &symbol,
+                                    const SyntaxTreeContext &context) {
+  verible::matcher::BoundSymbolManager manager;
+  if (valid_input_suffixes.empty() && valid_output_suffixes.empty()) return;
+
+  // Types of assignments intended for DFF inputs
+  static const Matcher blocking_assignment_matcher = verible::matcher::AnyOf(
+      NodekContinuousAssignmentStatement(), NodekNetVariableAssignment(),
+      NodekNetDeclarationAssignment(), NodekAssignModifyStatement(),
+      NodekIncrementDecrementExpression());
+  if (blocking_assignment_matcher.Matches(symbol, &manager)) {
+    HandleBlockingAssignments(symbol, context);
+    return;
+  }
+
+  // From this point, ignore everything that isn't a nonblocking assignment
+  // inside an always_ff block
+  if (!NodekNonblockingAssignmentStatement().Matches(symbol, &manager) ||
+      !context.NearestParentMatching(
+          [&](const verible::SyntaxTreeNode &node) -> bool {
+            return AlwaysFFMatcher().Matches(node, &manager);
+          })) {
+    return;
+  }
+
+  // Waive if this particular nonblocking assignment is inside an if block
+  // related to a reset signal, as we are not capable of determining
+  // reset-values from analyzing the source code.
+  const verible::SyntaxTreeNode *waive = context.NearestParentMatching(
+      [&](const verible::SyntaxTreeNode &node) -> bool {
+        if (!node.MatchesTag(NodeEnum::kIfClause)) {
+          return false;
+        }
+
+        const verible::SyntaxTreeNode *if_header = GetIfClauseHeader(node);
+        if (!if_header) return false;
+        const verible::Symbol *s = GetIfHeaderExpression(*if_header);
+
+        if (!s) return false;
+        absl::string_view paren_str =
+            absl::StripAsciiWhitespace(verible::StringSpanOfSymbol(*s));
+
+        // EXACT matching w.r.t waive_conditions. Substring checking isn't
+        // really appropriate because we would have to check for tricky stuff
+        // like negation, negation around parenthesis...
+        return std::find(waive_conditions.cbegin(), waive_conditions.cend(),
```

Comment:
I'd use `absl::StrContains()` here for readability.

Reply:
Sorry if I'm being dense but I don't follow, what do you suggest exactly?

Reply:
Looks like you're looking if a substring matches.
The `absl::StrContains()` function (include `absl/strings/match.h`) provides this in a way that improves readability without need of iterators and such.

Reply:
But `absl::StrContains` only searches on strings, right? and I'm searching for equality in a vector of strings

Reply:
ah ok, I should read more closely :) You're right.